distributed/deploy/adaptive.py

```diff
@@ -80,13 +132,13 @@ def _adapt(self):
         self._adapting = True
         try:
             if self.should_scale_up():
```

Comment:
Could you say a bit more about your concern? Or if I can try to guess, the concern is that you get into a situation of competing `scale_up` and `scale_down` calls?

The `should_scale_down` method may not be the clearest name, it's more like `should_check_with_the_scheduler_if_there_are_any_workers_to_scale_down`. It's that "check with the scheduler" part that matters. The default is always `True`, so always check. And then the default behavior of the scheduler is to retire any workers that are idle.

The new behavior should be identical to the previous behavior. It's just adding a point where a subclass can overrride.